website/docs/r/s3_bucket_object.html.markdown

```diff
@@ -68,8 +68,9 @@ The following arguments are supported:
 
 * `bucket` - (Required) The name of the bucket to put the file in.
 * `key` - (Required) The name of the object once it is in the bucket.
-* `source` - (Required) The path to the source file being uploaded to the bucket.
-* `content` - (Required unless `source` given) The literal content being uploaded to the bucket.
+* `source` - (Required unless `content` or `content_base64` is set) The path to a source file to upload to the bucket.
+* `content` - (Required unless `source` or `content_base64` is set) Literal content to upload into the bucket.
```

Comment:
Instead of `Literal content` here should we say `String content`/`UTF8 content`?

Reply:
"Literal" here was trying to distinguish from "filename" and "base64" rather than making a statement about what format the content is in. That is, I was trying to explain how `source` is different from `content` to someone who might not be familiar with either yet.

I'm not sure that "string" gets to that meaning because the base64 version and the filename are also both strings, just with different interpretations of their values.

How about this?

* `source` - The path to a file that will be read and uploaded as raw bytes for the object content.
* `content` - Literal string value to use as the object content, which will be uploaded as UTF-8-encoded text.
* `content_base64` - Base64-encoded data that will be decoded and uploaded as raw bytes for the object content.

Reply:
Works for me 👍